1. Phenotype_results.xls: Seven quantitative traits for early maturity, comprising flowering time (FT), the period from the first flower blooming to the first boll opening (FBP), Whole growth period (WGP), yield percentage before frost (YPBF), node of the first fruiting branch (NFFB), height of the node of the first fruiting branch (HNFFB) and plant height (PH) used for GWAS analysis. 
2. pheatmap.fpkm: Expression pattern of flowering time-related genes at square, 5 dps, 10 dps, 15 dps and 20 dps.
3. Sanger_Sequencing.tar.gz: anger sequencing for validate the accuracy of the SNP results.
4. 436.reseq.conf: A script used for circos plot.
5. bcftools_calling.sh: A script used for variations calling by bcftools.
6. gatk_calling.sh: A script used for variations calling by GATK.
7. get4D_loci.py: A script used for getting four-fold degenerate sites in the upland cotton genome.
8. map.R: A script used for plot world map in the Figure 1.
9. mapping.sh: A script used for reads mapping in the  upland cotton genome.
10. markdup.sh: A script used for mark duplication in the bam files.
11. pca.R: A script used for PCA plot.
12. readsQC.sh: A script used for raw reads quailty control.
13. sam2bam.sh: A script used for convert sam to bam.
14. sortbam.sh: A script ursd for sort reads in the bam file.
15. structure.sh: A script used for sturcture analysis.
16. tree.R: A script used for visualization of phylogenetic trees
17. upload2NCBI.sh: A script used for upload RNAseq and resequencing data to NCBI.
18. variation_annotation.sh:  A script ursd for variation annotation.
19. variation_filter.sh: A script ursd for SNPs and Indels filtering.
20. DPS_0_20.FPKM.xls: RNA-Seq analysis at five developmental stages as follows: (1) square; (2) 5 days post square (5 DPS); (3) 10 DPS; (4) 15 DPS; (5) 21 DPS between ‘CRI50’ and ‘TM-1’.
21. gwas_analysis.sh: A script used for association analysis for early maturity traits.
22. Manhattan_plot.py: A script used for plot Manhattan for GAWS results for early maturity traits.
23. RNAseq_analysis.sh: A script used for calculating all gene expression leves by RNAseq.
24. blup.R: A script used for calculating BLUP value for each early maturity-related traits.
25. plot_LDheatmap.py: A script used for plot LDheatmap.
26. PCA.sh: A script used for PCA analysis for all accessions.
